File: src/main/java/com/github/tanokun/tanorpg/player/quest/action/QuestGiveExpAction.java
package com.github.tanokun.tanorpg.player.quest.action;

import com.github.tanokun.tanorpg.TanoRPG;
import com.github.tanokun.tanorpg.player.Member;
import com.github.tanokun.tanorpg.util.command.CommandContext;
import org.bukkit.Bukkit;

public class QuestGiveExpAction implements Action {
    int exp;

    public QuestGiveExpAction(CommandContext cc){
        exp = Integer.valueOf(cc.getArg(0, "1"));
    }

    @Override
    public void execute(Member m) {
        Bukkit.getScheduler().runTask(TanoRPG.getPlugin(), () -> m.addHasEXP(exp));
    }
}

File: src/main/java/com/github/tanokun/tanorpg/player/quests/actions/QuestGiveExpAction.kt
package com.github.tanokun.tanorpg.player.quests.actions

import com.github.tanokun.tanorpg.TanoRPG
import com.github.tanokun.tanorpg.player.Member
import com.github.tanokun.tanorpg.player.quests.utils.annotation.ActionData
import com.github.tanokun.tanorpg.util.regex.RegexMatcher
import org.bukkit.Bukkit

@ActionData("giveExp", "<Int:exp>", "giveExp [exp]", "playerに[exp] exp渡します")
class QuestGiveExpAction(regexMatcher: RegexMatcher, parameters: String): QuestAction {
    private var exp = 0

    init {
        regexMatcher.matchResult(parameters) {
            exp = it.get("exp", 0)
        }
    }

    override fun run(member: Member) {
        Bukkit.getScheduler().runTask(TanoRPG.plugin, Runnable { member.addHasEXP(exp.toLong()) })
    }
}